Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs

```diff
@@ -556,7 +556,7 @@ private void UpdateGazeProvider()
         {
             // The gaze hit result may be populated from previous raycasts this frame, only recompute
             // another raycast if it's not populated
-            if (gazeHitResult == null)
+            if (gazeHitResult == null && gazeProviderPointingData != null)
```

Comment:
Not sure understanding why this is necessary? Seems like UpdateGazeProvider() won't be called at all if gazeProviderPointingData or it's pointer is null? See L 542 above

Reply:
While that's true at the moment, I figured it'd better be safe than sorry since nothing is preventing this method from being called elsewhere in the script.

Reply:
Sure, and I guess I would say two things.

1) Remove the gazeProviderPointingData null check that is wrapped around the call to UpdateGazeProvider()

2) Concerning Julia's point, I feel like there are a lot of items that could be null in FocusProvider (as well as of course in almost every component) and adding a null check test for just one doesn't seem like it delivers a lot. It would be better to add test holistically and maybe even for the items we expect likely will be null in some instance

Reply:
I think this is a unique thing in FocusProvider though, where it being null is actually "okay". We want to support the scenario where the gaze provider is optional, and right now that's not possible.

Reply:
Removed the duplicate null check